pyrobosim/pyrobosim/core/robot.py

```diff
@@ -51,11 +64,22 @@ def __init__(
         """
         # Basic properties
         self.name = name
-        self.set_pose(pose)
         self.radius = radius
         self.height = height
         self.color = color
 
+        # Dynamics properties
+        self.dynamics = RobotDynamics2D(
+            robot=self,
+            init_pose=pose,
+            max_linear_velocity=max_linear_velocity,
+            max_angular_velocity=max_angular_velocity,
+            max_linear_acceleration=max_linear_acceleration,
+            max_angular_acceleration=max_angular_acceleration,
+        )
+        self.dynamics.robot = self
```

Comment:
The `robot` and `pose`  attributes would already be set during the construction of `RobotDynamics2D` object right.
Do we have to explicitly initialise it here?

Reply:
Oh yeah, great catch! This was left over from before I had a proper constructor. Fixed.